### PR TITLE
fix(chat): fix displaying endpoints and environmental variable in review request (Issue #2521)

### LINKED ISSUE #2521: [Code app +Publication] Fields Endpoints and Environmental variable are not displayed on review form when admin review publication request
### EPAM AI DIAL chat version

0.20.0

### How to reproduce

Steps:
1. by user create code app
2. input Environmental variable and Endpoint Chat Completion
3. Save app
4. Create publication request for Code app
5. by admin click on publication request name for code app in Approve required section
6. click "Go to a review"

### Actual result

Fields Endpoint and Environmental variable are not displayed
image (52)

### Expected result

Fields Endpoint and Environmental variable should be available for Admin to review for publication requests for code apps
https://www.figma.com/design/lwL5jneveZczAj5RLcLFH2/DIAL-Chat?node-id=13596-45887&t=JsXuzYyonQ8W4g5N-1
### Confidential information

- [X] I confirm that do not share any confidential information

### PR DESCRIPTION
**Pull request by denys-kolomiitsev**
**Description:**

Fix displaying fields "Endpoints" and "Environmental variable" in review form when admin review publication request

Issues:

- Issue #2521 

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2559.nightly-test.deltixhub.io
>E2E tests status: failed

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx [hunk 1]
```diff
@@ -118,16 +121,31 @@ export function ReviewApplicationDialogView() {
...
+              <span className="w-[122px] text-secondary">
+                {t('Completion URL:')}
+              </span>
+              <span className="max-w-[414px] break-all text-primary">
+                {application.completionUrl}
+              </span>
+            </div>
+          )}
+        {application?.function?.mapping &&
+          !isEmpty(application?.function?.mapping) && (
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {!isEmpty(application?.function?.mapping) && (
```

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx [hunk 2]
```diff
@@ -118,16 +121,31 @@ export function ReviewApplicationDialogView() {
...
+            <ReviewApplicationPropsSection
+              label="Endpoints"
+              appProps={application.function.mapping}
+              propsNames={FEATURES_ENDPOINTS_NAMES}
+            />
+          )}
+        {application?.function?.env && !isEmpty(application?.function?.env) && (
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {!isEmpty(application?.function?.env) && (
```

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx (hunk 1 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]: Thank you for the suggestion!
Unfortunately build is failing without this.

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx (hunk 2 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]: Thank you for the suggestion!
Unfortunately build is failing without this.

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/535081)

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/535097)

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2559.nightly-test.deltixhub.io
>E2E tests status: failed

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2559.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/ReviewApplicationPropsSection.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export const ReviewApplicationPropsSection = ({
+  label,
+  appProps,
+  propsNames,
+}: SectionProps) => {
+  const { t } = useTranslation(Translation.Chat);
+  return (
+    <div className="flex gap-4">
+      <span className="w-[122px] text-secondary">{t(`${label}`)}:</span>
```
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationPropsSection.tsx]:
```suggestion
      <span className="w-[122px] text-secondary">{t(label)}:</span>
```

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx (hunk 2 above)
Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {application?.function?.env && !isEmpty(application.function.env) && (
```

Comment (Alexander-Kezik) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {application?.function?.env && !isEmpty(application.function.env) && (
```

**Review by Alexander-Kezik — DISMISSED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2559.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx (hunk 2 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]: Thank you for the suggestion!
Updated as suggested.

Comment (denys-kolomiitsev) [reply apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]: Thank you for the suggestion!
Updated as suggested.

File: apps/chat/src/components/Chat/Publish/ReviewApplicationPropsSection.tsx (hunk 3 above)
Comment (denys-kolomiitsev) [reply apps/chat/src/components/Chat/Publish/ReviewApplicationPropsSection.tsx]: Thank you for the suggestion!
Updated as suggested.

**Review by Alexander-Kezik — DISMISSED** [review]

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx [hunk 4]
```diff
@@ -11,10 +11,13 @@ import { Translation } from '@/src/types/translation';
...
 import { useAppSelector } from '@/src/store/hooks';
 
-import { FEATURES_ENDPOINTS } from '@/src/constants/applications';
+import { FEATURES_ENDPOINTS_NAMES } from '@/src/constants/applications';
 
 import { ModelIcon } from '../../Chatbar/ModelIcon';
 import { PublicationControls } from './PublicationChatControls';
+import { ReviewApplicationPropsSection } from './ReviewApplicationPropsSection';
+
+import { isEmpty } from 'lodash-es';
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
import isEmpty from 'lodash-es/isEmpty';
```

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx [hunk 5]
```diff
@@ -118,16 +121,31 @@ export function ReviewApplicationDialogView() {
...
+              <span className="max-w-[414px] break-all text-primary">
+                {application.completionUrl}
+              </span>
+            </div>
+          )}
+        {application?.function?.mapping &&
+          !isEmpty(application.function.mapping) && (
+            <ReviewApplicationPropsSection
+              label="Endpoints"
+              appProps={application.function.mapping}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {!isEmpty(application?.function?.mapping) && (
            <ReviewApplicationPropsSection
              label="Endpoints"
              appProps={application?.function?.mapping!}
```

File: apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx [hunk 6]
```diff
@@ -118,16 +121,31 @@ export function ReviewApplicationDialogView() {
...
+              propsNames={FEATURES_ENDPOINTS_NAMES}
+            />
+          )}
+        {application?.function?.env && !isEmpty(application.function.env) && (
+          <ReviewApplicationPropsSection
+            label="Environment variables"
+            appProps={application.function.env}
```
Comment (IlyaBondar) [inline apps/chat/src/components/Chat/Publish/ReviewApplicationDialogView.tsx]:
```suggestion
        {!isEmpty(application?.function?.env) && (
          <ReviewApplicationPropsSection
            label="Environment variables"
            appProps={application?.function?.env!}
```

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/535291)

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>- **Deployment failed**: [failed](https://gitlab.deltixhub.com/Clients/deltix/nightly-test/dial/review/-/pipelines/535301)

**Review by IlyaBondar — APPROVED** [review]

**Comment by denys-kolomiitsev** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-2559.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**